pytests/test_html.py

```diff
@@ -1,8 +1,40 @@
-
 from subprocess import run
 from helper import get_html_paths
 
+from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.support.wait import WebDriverWait
+import pytest
+from helper import get_css_absolute_paths, fd_input
+
+def test_html_w3c(driver: WebDriver):
+    files = get_html_paths()
+    for file in files:
+        driver.get("https://validator.w3.org/#validate_by_upload")
+        file_input = WebDriverWait(driver, 10).until(
+            lambda d: d.find_element(By.ID, "uploaded_file")
+        )
+
+        file_input.send_keys(file)
+        file_input.submit()
+
+        WebDriverWait(driver, 10).until(
+            lambda d: d.find_element(By.ID, "results")
+        )
+
+        # if the validation failed, the "success" class won't be present
+        # instead, the "failure" class will be present
+        try:
+            WebDriverWait(driver, 10).until(
+                lambda d: d.find_element(By.CLASS_NAME, "success")
+            )
+        except:
+            fd_input("Press enter to continue...")
+            pytest.fail(f"W3C html validation failed for {file}")
+
 
 def test_html_is_valid():
     cmd = f"npx html-validate {' '.join(get_html_paths())}"
```

Comment:
Tego testu bym nie zmieniał. W testach css użyłem `fd_input("Press enter to continue...")` po to, żeby móc ręcznie zobaczyć błąd z serwisu (inaczej mi się nie wyświetlał). Natomiast w tym teście to dobrze jak po prostu failujue, odpowiednia informacja wyświetla mi się na ekran i tak.

Reply:
cofnąłem zmianę